rir/src/interpreter/interp.cpp

```diff
@@ -1514,11 +1515,11 @@ SEXP evalRirCode(Code* c, Context* ctx, SEXP* env, const CallContext* callCtxt,
             CallContext call(c, ostack_at(ctx, n), n, ast,
                              ostack_cell_at(ctx, n - 1), getenv(), ctx);
             res = doCall(call, ctx);
-            ostack_popn(ctx, n + 1);
+            ostack_popn(ctx, call.nargs + 1);
```

Comment:
Why don't we use `call.nargs` or `n` in both cases?

Reply:
n is the supplied number of args. call.nargs is the actual number (after fixed up missings). so before the call we have n on the stack, after the call we have call.nargs on the stack.

Reply:
`n` is read from the bytecode. It is the number of args the function expects. And call.nargs is finally patched so that it contains the actual number of arguments too (previously it contained only the actual syntactic arguments passed). So If I understand this correctly they are the same and it is confusing to pop for `call.nargs` but assert on `n`. If I am wrong in the premise just ignore the comment and merge.

Reply:
```
n = how many args did the user supply
call.nargs = now many args are passed
fun->signature().nargs = how many args does the function want
```

* if we stuffed the passed arguments with missing, then `call.nargs == fun->signature().nargs`.
* if we did not stuff passed arguments with missing, because they are not passed on the stack, then `n == call.nargs`.